### PR TITLE
Feature: Added token support to CircleCI badges

### PR DESCRIPTION
**Pull request by krak3n**
This PR adds CircleCI token support in the same vein as the token feature on CodeCov allowing serving of private CircleCI builds.

See regex here: http://regexr.com/3b2v7

### CODE REVIEW

File: server.js [hunk 1]
```diff
@@ -3302,22 +3302,35 @@ cache(function(data, match, sendBadge, request) {
...
+  queryParams = {};
+  queryParams['limit'] = 1;
+  queryParams['filter'] = 'completed';
+
+  // Custom Banch if present
   if(branch != null) {
     apiUrl += "/tree/" + branch;
   }
-  apiUrl +=
-    '?circle-token=0a5143728784b263d9f0238b8d595522689b3af2'
```
Comment (espadrine) [inline server.js]: I'm confused. Was having a default token never necessary?

Here's what I was told back then: https://github.com/badges/shields/pull/348#discussion_r22603858

Comment (krak3n) [inline server.js]: The token is used to query the circle REST API and isn't be needed for public project build states. If the project is private then you need a token to access the REST API to get the build state of the project.

So the example project is public and thus works without a token:

```
curl https://circleci.com/api/v1/project/BrightFlair/PHP.Gt\?limit\=1\&filter\=completed -H "Accept:application/json"
```

Comment (espadrine) [inline server.js]: So, what if the example project is private? Will existing badges stop working?

**Comment by maletor** [comment]
Perfecto!

File: server.js (hunk 1 above)
Comment (krak3n) [inline server.js]: Yes it they would.

Comment (krak3n) [inline server.js]: @espadrine at the moment I think that the token is ignored, public projects wouldn't break if the example repo became private. For example I can put a random circle token on the end and the API responds normally:

```
curl https://circleci.com/api/v1/project/BrightFlair/PHP.Gt\?limit\=1\&filter\=completed\&circle-token\=foo -H "Accept:application/json"
```

Basically all this PR does is remove the ignored token by default and allow the user to pass their own token in for private projects.

**Comment by espadrine** [comment]
Thanks for contributing!

**Pull request merged**